docs/spec.md

```diff
@@ -0,0 +1,114 @@
+Universe Spec
+=============
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## MUST define a `package.json`
+
+Schema: [package-schema.json](../repo/meta/schema/package-schema.json)
+
+### MUST specify a unique `name` that is not used by any other package in the repo
+
+### MUST specify a meaningful `version` that allows for targeted debugging
+
+### MUST provide a meaningful `description` that will give the user insight into the service
+
+### MUST provide a set of meaningful `tags` that will allow a user to search for the service
+
+### MUST provide a `maintainer` property
+
+### MUST specify a `images` object
+
+1. All images SHOULD be free of visual artifacts such as JPG compression artifacts
+1. All icons and schreenshots SHOULD provide a sibling resource for retina-ready displays (See [Retina Display Support](#Retina-Display-Support))
+1. MUST provide URL for `icon-small` property (48px W x 48px H)
+1. MUST provide URL for `icon-medium` property (96px W x 96px H)
+1. MUST provide URL for `icon-large` property (256px W x 256px H)
+1. MAY provide a set of `screenshots` (1200px W x 675px H)
+
+### MAY specify a meaningful `preInstallNotes` property that will be displayed to the user prior to performing an install
+
+### SHOULD specify a meaning `postInstallNotes` property that will be displayed to the user after performing an install
+
+Good items to include are:
+* Link to Service Documentation
+* Link to Issue tracker for bug reports
+
+
+## MUST define a `config.json`
+
+Schema: [config-schema.json](../repo/meta/schema/config-schema.json)
+
+### MUST namespace all properties
+
+*A single `options.json` should be able to be used for an entire DCOS Cluster*
+
+### MUST use hierarchical definition format
+
+1. MUST set `additionalProperties` to `false` for all nested objects
```

Comment:
Not all nested objects. I think we want this only on the package specific JSON object. For example I don't think we want it to be `false` for the Mesos' nested object.